src/routes/docs/tutorials/nextjs/step-5/+page.markdoc

```diff
@@ -0,0 +1,278 @@
+
+We have created a new API endpoint that will add a new expense to the database. The endpoint will be available at `/api/expense`.
+
+Let's use the new API. In the `/src/app/index.js` file, add the following code to the `addExpense` function. This function will add a new expense to the database.
+
+```js
+"use client";
+
+import React, { useState } from "react";
+import { useCookie } from "next-cookie";
+import toast from "react-hot-toast";
+
+const Home = (props) => {
+  const [popup, setPopup] = useState(null);
+  const cookie = useCookie(props.cookie);
+  const router = useRouter();
+  const jwt = cookie.get("jwt");
+  const userId = cookie.get("userid");
+
+  const createExpense = async (title, amount, type) => {
+    const response = await fetch("/api/expense", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        jwt,
+      },
+      body: JSON.stringify({
+        userId: userId,
+        title: title,
+        amount: amount,
+        type: type,
+      }),
+    });
+
+    if (response.status === 200) {
+      toast.success("Saved successfully");
+      setPopup(null);
+    }
+  };
+
+  return (
+    <>
+      {popup?.type === "new" && (
+        <CreateExpensePopup 
+          setPopup={setPopup} 
+          createExpense={createExpense} 
+        />
+      )}
+      <div className="container u-flex-vertical u-gap-12 u-cross-center">
+        <h1 className="heading-level-1 font-bold">Your expenses</h1>
+        <button
+          className="button"
+          onClick={() => setPopup({ type: "new", data: null })}
+        >
+          <span className="icon-plus" aria-hidden="true"></span>
+          <span className="text">New Expense</span>
+        </button>
+      </div>
+    </>
+  );
+};
+```
+
+We are rendering a new component `CreateExpensePopup` when the user clicks on the `New Expense` button. The `CreateExpensePopup` component will be used to create a new expense. 
+
+Let's create the `CreateExpensePopup` component. Create a new file `/src/components/CreateExpensePopup.js` and add the following code.
+
+```js
+"use client";
+
+import React, { useState } from "react";
+import toast from "react-hot-toast";
+
+const CreateExpensePopup = ({ setPopup, createExpense }) => {
+  const [item, setItem] = useState({
+    title: "",
+    amount: "",
+    type: "",
+  });
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!item.title) {
+      return toast.error("Title is required");
+    }
+    if (!item.amount) {
+      return toast.error("Amount is required");
+    }
+    if (!item.type) {
+      return toast.error("Type is required");
+    }
```

Comment:
This should probably be a concern of `createExpense`. In the case there's an error there, you can catch the error here, and use a toast. Especially because other errors may heppen, e.g. a network failure